src/proj_4D_api.c

```diff
@@ -381,6 +381,111 @@ char*  proj_rtodms(char *s, double r, int pos, int neg) {
 }
 
 
+/*************************************************************************************/
+static int pj_cs2cs_emulation_setup (PJ *P) {
+/**************************************************************************************
+If any cs2cs style modifiers are given (axis=..., towgs84=..., ) create the 4D API
+equivalent operations, so the preparation and finalization steps in the pj_inv/pj_fwd
+invocators can emulate the behaviour of pj_transform and the cs2cs app.
+**************************************************************************************/
+    PJ *Q;
+    paralist *p;
+    char def[1000];
+    if (0==P)
+        return 0;
+
+    /* Don't recurse when calling proj_create (which calls us back) */
+    if (pj_param_exists (P->params, "break_cs2cs_recursion"))
+        return 1;
+
+    /* Swap axes? */
+    p = pj_param_exists (P->params, "axis");
+    /* p->param is a *string array*, not a *pointer to char*, */
+    /* so we do not check here and below whether it is non-null */
+    if (p && strcmp ("enu", p->param)) {
```

Comment:
At this point I believe `P->axis` can be expected to contain valid data if it is set. So no need for the `strcmp`.

Reply:
The strcmp returns true (i.e. non-null) if  *(p->param) != "enu", so it's a way to avoid calling axisswap if the data are already in the expected order. I agree that the data must be valid at this point, so this is not a validation, just a cheap, optimization, which may deserve a comment to explain...

Reply:
Ah of course. Makes sense.